library/src/data/index.js

```diff
@@ -41,20 +41,236 @@ window.CoNDeT.data = {
+    };
+
+    addColumn = function (tableId, column) {
+      // column = {condition: "..." albo null, decision: "..." albo null}
+      var newState = copyTables(this.state);
+      var editTable = newState[tableId];
+
+      if (column.condition != null) {
+        editTable.columns.conditions.push(column.condition);
+      }
+      if (column.decision != null) {
+        editTable.columns.decisions.push(column.decision);
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    removeColumn = function (tableId, column) {
+      // column = {condition: idCon, decision: idDec}
+      var newState = copyTables(this.state);
+      var editTable = newState[tableId];
+
+      removeColumnWithId(
+        editTable.columns.conditions,
+        editTable.rows,
+        "conditions",
+        column.condition
+      );
+      removeColumnWithId(
+        editTable.columns.decisions,
+        editTable.rows,
+        "decisions",
+        column.decision
+      );
+
+      this.stateManager.setState(newState);
+    };
+
+    changeOrder = function (tableId) {};
+
+    addConnection = function (tableId, rowId, secondTableId) {
+      var newState = copyTables(this.state);
+      var editTable = newState[tableId];
+
+      for (var i = 0; i < editTable.rows.length; i++) {
+        if (editTable.rows[i].id == rowId) {
+          editTable.rows[i].connections.push(secondTableId);
+        }
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    removeConnection = function (tableId, rowId, secondTableId) {
+      var newState = copyTables(this.state);
+      var editTable = newState[tableId];
+
+      for (var i = 0; i < editTable.rows.length; i++) {
+        for (var j = 0; j < editTable.rows[i].connections.length; j++) {
+          if (
+            editTable.rows[i].id == rowId &&
+            editTable.rows[i].connections[j] == secondTableId
+          ) {
+            editTable.rows[i].connections.splice(j, 1);
+          }
+        }
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    replaceInArray = function (toReplace, newValues) {
+      if (newValues != []) {
+        for (var i = 0; i < newValues.length; i++) {
+          toReplace[i] = newValues[i];
+        }
+      }
+    };
+
+    removeColumnWithId = function (columns, rows, type, id) {
+      if (id != null && id <= columns.length) {
+        columns.splice(id, 1);
+        for (var i = 0; i < rows.length; i++) {
+          if (type == "conditions") {
+            for (var j = 0; j < rows[i].conditions.length; j++) {
+              if (rows[i].conditions[j][0] == id) {
+                rows[i].conditions.splice(j, 1);
+              }
+            }
+          } else if (type == "decisions") {
+            for (var j = 0; j < rows[i].decisions.length; j++) {
+              if (rows[i].decisions[j][0] == id) {
+                rows[i].decisions.splice(j, 1);
+              }
+            }
+          }
+        }
+      }
```

Comment:
Jeżeli masz parametr type, to możesz zrobić takie coś:
```
for (var i = 0; i < rows.length; i++) {
  for (var j = 0; j < rows[i][type].length; j++) {
    if (rows[i][type][j][0] !== id) continue;
    rows[i][type].splice(j, 1);
  }
}
```
Bo obiekty w JS są tak naprawdę [mapami](https://en.wikipedia.org/wiki/Associative_array) (czy tam słownikami jak się mówi w Pythonie, albo tablicami asocjacyjnymi jak w PHP)